libmscore/read301.cpp

```diff
@@ -336,5 +337,183 @@ Score::FileError MasterScore::read301(XmlReader& e)
       return FileError::FILE_NO_ERROR;
       }
 
-}
+//------------------------------------------------------------------------------
+//   MStyle::spatium301() - For reading 3.01 and earlier file formats, where the
+//                          spatium is stored in floating-point millimeters.
+// These four hard-coded values are the only values that MuseScore ever stored
+// for the default spatium value in millimeters:
+//  1.76389 = SPATIUM20 in millimeters, rounded by Qt when converted to a string
+//  1.764   = 1.76389 rounded to 3 decimals by pagesettings.ui
+//  1.7526  = SPATIUM20 in inches, rounded to 3 decimals, and converted to mm
+//  1.753   = 1.7526 rounded to 3 decimals by pagesettings.ui
+// These four values do not convert cleanly to the default spatium value of
+// SPATIUM20. This code sets their value to SPATIUM20, instead of converting.
+//------------------------------------------------------------------------------
+
+void MStyle::spatium301(XmlReader& e)
+      {
+      QString txt = e.readElementText();
+      if (txt == "1.76389" || txt == "1.764" || txt == "1.7526" || txt == "1.753")
```

Comment:
This might be good to convert some old files if you are sure that all `1.7526` in your files result exactly from this bug, but should this conversion rule be applied always?

Reply:
These values appear in new user files today too.
I see your concern that the user might actually set their spatium to something equivalent to the last two of the four values.  I can see if that's possible with the new code.  I doubt that it is, because these numbers are based on specific rounding being done.

Reply:
Duh!  The user can enter `1.753` directly in the dialog box, and I am excluding that from the range of possible values.  It's an unusual value to enter intentionally, but still it's bad practice to exclude it.  AFAICT it is not possible to generate and store a value of `1.7526` with this PR's code.

This is part of why I want to move towards storing integer values and then all of this can go away some day.  As of v4, it would be relegated to the 301 code only.

In MuseScore 3.1 beta and previous versions:
`1.7526` millimeters exists in any file where the user selects the Inches radio button in pagesettings.ui, leaves the spatium value at `0.069` inches, then clicks Apply or OK.
`1.753` value happens subsequently, when the user switches back to Millimeters, then clicks Apply or OK.  Or if they truly intend to have this spatium value, which is unlikely, but possible.

I don't see a way of converting old files, except via a compatibility build and a major version change.  mtest files could be converted, but then they would not be representing the real world situation, which is not good.

In terms of risk:  I believe this one exception is very low risk.  I think that only a few user files have the `1.753` value, because it requires the switch back and forth between Inches and Millimeters.  I also highly doubt that it exists anywhere due to the user inputting that value intentionally, without ever switching to Inches.  `1.750` is a common value.  `1.753` is a rounding error.  From what I've heard, the 3rd decimal place is generally for things like `1.875`, divisions by 8 that can put a 5 in the 3rd decimal slot.

Reply:
Or screw the old files.  Let them live on with their slightly off-default spatium values.  That's not a bad idea.

Reply:
I just commented out the comparisons to `1.7526` and `1.753`.  Then ctest ran without errors.  I do see one file that has `1.7526` in it here:
https://github.com/musescore/MuseScore/blob/40dc553a758211e84e3e0972c7802023d4c39f4e/mtest/libmscore/repeat/repeat14.mscx#L1399
But that file appears to have no diff in the test, so there's no change there.

So I'm going to remove these two comparisons, and you'll see the change in the next push.